yt_dlp/extractor/asobistage.py

```diff
@@ -0,0 +1,145 @@
+    _TESTS = [{
+        'url': 'https://asobistage.asobistore.jp/event/315passionhour_2022summer/archive/frame',
+        'info_dict': {
+            'id': '315passionhour_2022summer/archive/frame/edff52f2',
+            'title': '315passion_FRAME_only',
+            'live_status': 'was_live',
+            'thumbnail': r're:^https?://[\w.-]+/\w+/\w+',
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        },
+    }, {
+        'url': 'https://asobistage.asobistore.jp/event/idolmaster_idolworld2023_goods/archive/live',
+        'info_dict': {
+            'id': 'idolmaster_idolworld2023_goods/archive/live/3aef7110',
+            'title': 'asobistore_station_1020_serverREC',
+            'live_status': 'was_live',
+            'thumbnail': r're:^https?://[\w.-]+/\w+/\w+',
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        },
+    }, {
+        'url': 'https://asobistage.asobistore.jp/event/ijigenfes_utagassen/player/day1',
+        'only_matching': True,
+    }]
+
+    _TOKEN = ''
+
+    def _check_login(self, video_id):
+        check_login_json = self._download_json(
+            'https://asobistage-api.asobistore.jp/api/v1/check_login', video_id, expected_status=(400),
+            note='Checking login status', errnote='Unable to check login status')
+        error = traverse_obj(check_login_json, ('payload', 'error_message'), ('error'), get_all=False)
+
+        if not error:
+            return
+        elif error == 'notlogin':
+            self.raise_login_required()
+        else:
+            raise ExtractorError(f'Unknown error: {error!r}')
+
+    def _get_owned_tickets(self, video_id):
+        for url, name in [
+            ('https://asobistage-api.asobistore.jp/api/v1/purchase_history/list', 'ticket purchase history'),
+            ('https://asobistage-api.asobistore.jp/api/v1/serialcode/list', 'redemption history'),
+        ]:
+            yield from traverse_obj(self._download_json(
+                url, video_id, note=f'Downloading {name}', errnote=f'Unable to download {name}'),
+                ('payload', 'value', ..., 'digital_product_id', {str}))
+
+    def _real_initialize(self):
+        self._TOKEN = self._download_json(
+            'https://asobistage-api.asobistore.jp/api/v1/vspf/token', None,
+            note='Getting token', errnote='Unable to get token')
+
+    def _real_extract(self, url):
+        video_id, video_type_name = self._match_valid_url(url).group('id', 'type')
+        self._check_login(video_id)
+
+        webpage = self._download_webpage(url, video_id)
+
+        video_type_id = {
+            'archive': 'archives',
+            'player': 'broadcasts',
+        }.get(video_type_name)
+        if not video_type_id:
+            raise ExtractorError('Unknown video type')
```

Comment:
we should check if it's not a supported url before doing anything else, and include the unsupported type in the error output
```suggestion
        video_type_id = {
            'archive': 'archives',
            'player': 'broadcasts',
        }.get(video_type_name)
        if not video_type_id:
            raise ExtractorError(f'Unknown video type {video_type_name!r}')

        self._check_login(video_id)
        webpage = self._download_webpage(url, video_id)
```

Reply:
alternatively we could only match `archive|player` in the `_VALID_URL`